drivers/timer/ifx_cat1_lp_timer_pdl.c

```diff
@@ -0,0 +1,351 @@
+
+	k_spin_unlock(&lock, key);
+
+	/* Convert in hw cycles (keeping 32bit value) */
+	return (uint32_t)ret;
+}
+
+static void lptimer_isr(const struct device *dev)
+{
+	const struct ifx_cat1_lptimer_config *config = dev->config;
+	const struct ifx_cat1_lptimer_data *data = dev->data;
+
+	Cy_MCWDT_ClearInterrupt(config->reg_addr, (CY_MCWDT_CTR0 | CY_MCWDT_CTR1 | CY_MCWDT_CTR2));
+
+	/* Clear interrupt mask if set only from cyhal_lptimer_set_delay() function */
+	if (data->clear_int_mask) {
+		Cy_MCWDT_SetInterruptMask(config->reg_addr, 0);
+	}
+
+	if ((data->isr_instruction & LPTIMER_ISR_CALL_USER_CB_MASK) != 0) {
+		/* announce the elapsed time in ms */
+		uint32_t lptimer_value = Cy_MCWDT_GetCount(config->reg_addr, CY_MCWDT_COUNTER2);
+		int32_t delta_ticks = ((uint64_t)(lptimer_value - last_lptimer_value) *
+				       CONFIG_SYS_CLOCK_TICKS_PER_SEC) /
+				      config->clock_frequency;
+		sys_clock_announce(IS_ENABLED(CONFIG_TICKLESS_KERNEL) ? delta_ticks
+								      : (delta_ticks > 0));
+		last_lptimer_value = lptimer_value;
+	}
+}
+
+static int lptimer_init(const struct device *dev)
+{
+#define LPTIMER_CTRL (CY_MCWDT_CTR0 | CY_MCWDT_CTR1 | CY_MCWDT_CTR2)
+
+	const struct ifx_cat1_lptimer_config *config = dev->config;
+	struct ifx_cat1_lptimer_data *data = dev->data;
+	cy_rslt_t rslt = CY_MCWDT_BAD_PARAM;
+	cy_stc_mcwdt_config_t cfg = lptimer_default_cfg;
+
+	data->clear_int_mask = false;
+	data->isr_instruction = LPTIMER_ISR_CALL_USER_CB_MASK;
+
+	rslt = (cy_rslt_t)Cy_MCWDT_Init(config->reg_addr, &cfg);
+	if (rslt == CY_RSLT_SUCCESS) {
+		Cy_MCWDT_Enable(config->reg_addr, LPTIMER_CTRL, LPTIMER_RESET_TIME_US);
+	} else {
+		Cy_MCWDT_Disable(config->reg_addr, LPTIMER_CTRL, LPTIMER_RESET_TIME_US);
+		Cy_MCWDT_DeInit(config->reg_addr);
+	}
+
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQ(0, priority), lptimer_isr, DEVICE_DT_INST_GET(0),
+		    0);
+	irq_enable(DT_INST_IRQN(0));
+
+	return 0;
+}
+
+#define INFINEON_CAT1_LPTIMER_INIT(n)                                                              \
+                                                                                                   \
+	static struct ifx_cat1_lptimer_data ifx_cat1_lptimer_data##n;                              \
+                                                                                                   \
+	static const struct ifx_cat1_lptimer_config lptimer_cat1_cfg_##n = {                       \
+		.reg_addr = (MCWDT_STRUCT_Type *)DT_INST_REG_ADDR(n),                              \
+		.irq_priority = DT_INST_IRQ(n, priority),                                          \
+		.clock_frequency = DT_INST_PROP(n, clock_frequency),                               \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, lptimer_init, NULL, &ifx_cat1_lptimer_data##n,                    \
```

Comment:
No need to make the timer a device, its a singleton in Zephyr's view. Just a couple of static variables tied to the compilation unit for the hardware timer being used as the kernel timer.

Take a look at how [cortex_m_systick.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/timer/cortex_m_systick.c). It works by  SYS_INIT and a few static variables tied to the compilation unit, along with the appropriate sys_clock functions implemented.